### PR TITLE
Added delete using http post

### PR DESCRIPTION
**Pull request by mphprogrammer**
Added delete functionality, since the likelihood of changing http verb to delete from post is extremely slim.

### CODE REVIEW

**Comment by ayaleloehr** [comment]
Redirecting to @saneshark if that works for him?

File: lib/va_healthcare_messaging/client.rb [hunk 1]
```diff
@@ -74,7 +74,9 @@ def perform(method, path, params = nil, headers = nil)
     ## as JSON if the body represents a sucessful response.
     ###################################################################################################################
     def process_response_or_error
-      return @response if @response.status == 200 && @response.body.empty?
```
Comment (saneshark) [inline lib/va_healthcare_messaging/client.rb]: @mphprogrammer why not just use the header status? do they provide something differently for secure messaging?

Comment (mphprogrammer) [inline lib/va_healthcare_messaging/client.rb]: The put the word "Success" in the body. Was causing an exception in JSON.load.

File: .rubocop.yml [hunk 2]
```diff
@@ -14,3 +14,6 @@ Style/AccessorMethodName:
 # These specs have an Integer literal that is an id and underscores would look weird
 Style/NumericLiterals:
   Enabled: false
+
+Metrics/AbcSize:
+  Max: 18
```
Comment (saneshark) [inline .rubocop.yml]: So I think 15 is a pretty reasonable number for this. In my experience it's usually a good indicator that something can be broken out to into separate private methods which aids readability and would this heuristic as well. I'm going to assume this is related to `process_response_or_error` in client.rb -- I would look to possibly maybe create some private helper methods instead overriding this cop.

**Comment by saneshark** [comment]
@mphprogrammer made a couple relatively minor recommendations -- you can probably ignore those if you want as I think we're going to have to refactor all of this when we move it to vets-api anyway.

**Comment by mphprogrammer** [comment]
@saneshark The SM stuff is a little different when posting. For example, when posting a new message all attributes of the message are returned (e.g., the assigned message id ....). However, when deleting the word "Success" is returned, which tripped an exception in JSON.load.

Could we leave the AbcSize to 18 for the time being? I am working on attachments now, and when through, I'll circle back and refactor. Thanks!

**Comment by mphprogrammer** [comment]
If you'd rather not, let me know, and I'll refactor now. If its ok to let it go temporarily, feel free to merge it. I'll swing back to refactor in a separate branch.

**Comment by saneshark** [comment]
No, I think you're good. Client needs an abstraction for Request, and once thats there, this will be less of an issue. LGTM

**Pull request merged**